client/app/collections/scheduleitems.coffee

```diff
@@ -3,19 +3,27 @@ module.exports = class ScheduleItemsCollection extends Backbone.Collection
     model: require '../models/scheduleitem'
     comparator: (si1, si2) -> si1.getDateObject().diff si2.getDateObject()
 
+    # The calendar tag for all event was referencing the same instance
+    # in every event object. This is not the case anymore, so we have
+    # to check every event calendar in the a given calendars collection
+    # to verify the visibility
+    visibleItems: (calendars) ->
+        return _ @filter (item) ->
```

Comment:
Actually, I helped myself on this one about how to filter a collection in backbone, with this article: http://www.position-absolute.com/articles/filtering-a-collection-in-backbone-js/
The guy is wrapping his filter with an underscore call, without actually knowing why. As you said, the goal is just to get  a each method right after.

Because you asked me and I wasn't able to explain, I looked for a better answer, and this one seems a lot better because it's more explicit :
http://stackoverflow.com/questions/11762105/filter-backbone-collection-by-attribute-value/11762576#11762576

So I changed the code and will commit it.

Thanks for the information about the return statement, still learning coffee. :)